### PR TITLE
Declare models and SQL functions

### PR DESCRIPTION
**Pull request by ivanovaleksey**

### CODE REVIEW

File: migrations/2018-06-18-131228_create_abac_subject/up.sql [hunk 1]
```diff
@@ -0,0 +1,7 @@
+create table abac_subject (
+    inbound abac_attribute,
+    outbound abac_attribute,
+    -- type text,
```
Comment (manifest) [inline migrations/2018-06-18-131228_create_abac_subject/up.sql]: we don't need the comment

File: migrations/2018-06-18-135804_define_abac_object_target_function/up.sql [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+            where array[inbound] <@ _attrs
+        union
+        select (r.outbound).value, (r.outbound).key, (r.outbound).namespace_id
+            from target as t
+            inner join abac_object as r on r.inbound = (t.value, t.key, t.namespace_id) ::abac_attribute
+    )
+    select (unnest(_attrs)).*
+    union
+    select * from target
+$$ language sql immutable;
```
Comment (manifest) [inline migrations/2018-06-18-135804_define_abac_object_target_function/up.sql]: should be stable, here and everywhere

File: migrations/2018-06-18-131228_create_abac_subject/up.sql (hunk 1 above)
Comment (ivanovaleksey) [reply migrations/2018-06-18-131228_create_abac_subject/up.sql]: fixed

File: migrations/2018-06-18-135804_define_abac_object_target_function/up.sql (hunk 2 above)
Comment (ivanovaleksey) [reply migrations/2018-06-18-135804_define_abac_object_target_function/up.sql]: fixed

**Pull request merged**